packages/pg/lib/client.js

```diff
@@ -381,7 +381,11 @@ class Client extends EventEmitter {
     this.activeQuery.handleCommandComplete(msg, this.connection)
   }
 
-  _handleParseComplete(msg) {
+  _handleParseComplete() {
+    if (this.activeQuery == null) {
+      this.emit('error', new Error('Received parseComplete when not in parsing state'))
```

Comment:
Should this use `_handleErrorEvent`, since the connection is in an unpredictable state afterwards?

Reply:
I can see how using `_handleErrorEvent` might be a good idea here too, I think what @brianc communicated to me about this PR is that he thinks the issue is always related to an unexpected message arriving at a client which is in the idle state and so if that is the case it might not be required to do the full `_handleErrorEvent` but I can see how maybe doing full `_handleErrorEvent` is more robust.